test_reqs.py

```diff
@@ -41,17 +36,24 @@ def test_existing_requirement(requirements, mock_dist, testdir, monkeypatch):
     assert 'passed' in result.stdout.str()
 
 
-@pytest.mark.parametrize('requirements', [
-    'foo',
-    'Foo',
-    'foo==1.0',
-    'Foo==1.0',
+@pytest.mark.parametrize('requirements, dist', [
+    ('foo-bar', stub(project_name='foo-bar', version='1.0')),
+    ('foo-bar==1.0', stub(project_name='foo-bar', version='1.0')),
+    # Capitalization
+    ('Foo-bar', stub(project_name='foo-bar', version='1.0')),
+    ('foo-bar', stub(project_name='Foo-bar', version='1.0')),
+    # Periods
```

Comment:
These two feel very odd.  You probably know this better than me, but I am not sure that the canonicalisation being done is a) according to spec, or b) the same as pip / setuptools , or 3) doesnt conflict with the pypi package  registries use of dots.  The last one worries me more than the others, as I expect foo.bar and foo-bar can be two different packages.  if you are sure that isnt possible, then please disregard my concern - unfortunately I dont have time to figure out how to prove or disprove that with the new warehouse which didnt have comparable APIs as the old site for this type of analysis.

Reply:
They're all the same:

https://pypi.org/project/readme_renderer/
https://pypi.org/project/readme-renderer/
https://pypi.org/project/readme.renderer/

You can trust the `packaging` library, it is the codification of any packaging specification.